src/Components/templateWeather.tsx

```diff
@@ -11,15 +11,33 @@ interface Props {
 }
 
 export class TemplateWeather extends React.Component<Props> {
+  spinValue = new Animated.Value(0);
+
+  spin = () => {
+    this.spinValue.setValue(0);
+
+    Animated.timing(this.spinValue, {
+      toValue: 1,
+      duration: 500,
+      useNativeDriver: true,
+    }).start();
+  };
+
   render() {
     const forecast: ForecastData = this.props.forecastElement;
+    const rotate = this.spinValue.interpolate({
```

Comment:
You can remove this func from render right ?

Basically, 'render' function is called many times (everytime props change for example)
so you don't want to put unnecessary stuff in it because it will create unnecessary memory being taken